### PR TITLE
Make wrong deposit processable

### PR DESCRIPTION
**Pull request by vtleonardo**
This PR add a fix required to sync to the production nodes, i.e to digest the deposit with the wrong chainID

### CODE REVIEW

File: application/utxohandler/utxohandler.go [hunk 1]
```diff
@@ -760,15 +760,17 @@ func (ut *UTXOHandler) addOneFastSync(txn *badger.Txn, utxo *objs.TXOut) error {
...
+			value, err := utxo.Value()
+			if err != nil {
+				utils.DebugTrace(ut.logger, err)
+				return err
+			}
+			err = ut.valueIndex.Add(txn, utxoID, owner, value)
+			if err != nil {
+				utils.DebugTrace(ut.logger, err)
+				return err
+			}
```
Comment (diligentis) [inline application/utxohandler/utxohandler.go:773]:
```suggestion
    // Workaround for a historical deposit that was on the incorrect chain. Required to sync production nodes.
		if !utxo.IsDeposit() {
		  break
		}
	  value, err := utxo.Value()
	  if err != nil {
		  utils.DebugTrace(ut.logger, err)
		  return err
	  }
	  err = ut.valueIndex.Add(txn, utxoID, owner, value)
	  if err != nil {
		  utils.DebugTrace(ut.logger, err)
		  return err
```

File: application/utxohandler/utxohandler.go [hunk 2]
```diff
@@ -825,6 +827,43 @@ func (ut *UTXOHandler) StoreSnapShotStateData(txn *badger.Txn, utxoID, preHash,
 		}
 		return errorz.ErrInvalid{}.New(fmt.Sprintf("utxoHandler.StoreSnapShotStateData; utxoID does not match calcUtxoID; utxoID: %x; calcUtxoID: %x calcTxHash: %x TxOutIdx: %v", utxoID, calcUtxoID, calcTxHash, utxoIdxOut))
 	}
+	if utxo.IsDeposit() {
```
Comment (diligentis) [inline application/utxohandler/utxohandler.go:830]: Same thing here.

**Review by diligentis — CHANGES_REQUESTED** [review]
Workaround should be commented.

File: application/utxohandler/utxohandler.go (hunk 1 above)
Comment (vtleonardo) [reply application/utxohandler/utxohandler.go:777]: Addressed!

File: application/utxohandler/utxohandler.go (hunk 2 above)
Comment (vtleonardo) [reply application/utxohandler/utxohandler.go:839]: Addressed!

**Pull request merged**